src/slate-toolbar/modal-dialog.tsx

```diff
@@ -34,29 +54,155 @@ export const ModalDialog: React.FC<IProps> = (props) => {
       document.removeEventListener("mousedown", onMouseDown, true);
       document.removeEventListener("touchstart", onMouseDown, true);
     };
-  }, []);
+  }, [isMenuOpen]);
+
+  // auto-focus the first input
   const input1Ref = useRef<HTMLInputElement>(null);
   useEffect(() => {
-    setTimeout(() => input1Ref?.current?.focus());
+    if (input1Ref.current) {
+      setTimeout(() => input1Ref.current?.focus());
+    }
   }, []);
 
   // handlers
   const handleCancelClick = () => {
-    props.onClose(null);
+    props.onClose();
   };
   const handleOkClick = () => {
-    props.onClose(inputValues);
+    props.onClose(fieldValues);
   };
   const handleKeyDown = (e: React.KeyboardEvent<HTMLDivElement>) => {
-    if ((e.key === "Enter") || (e.keyCode === 13)) return handleOkClick();
-    if ((e.key === "Escape") || (e.keyCode === 27)) return handleCancelClick();
+    if (!isMenuOpen) {
+      if ((e.key === "Enter") || (e.keyCode === 13)) return handleOkClick();
+      if ((e.key === "Escape") || (e.keyCode === 27)) return handleCancelClick();
+    }
   };
-  const handleValueChange = (index: number) => (e: any) => {
-    const newArr = [...inputValues];
-    newArr[index] = e.target.value;
-    setInputValue(newArr);
+  const handleValueChange = (name: string, value: string) => {
+    setFieldValues(_values => {
+      const newValues = cloneDeep(fieldValues);
+      newValues[name] = value;
+      return newValues;
+    });
   };
 
+  function renderInput(field: IField, index: number, inline: boolean) {
+    const inputType = field.type === "checkbox" ? field.type : "text";
+    const inValue = field.type === "checkbox"
+                      ? undefined
+                      : fieldValues[field.name] || "";
+    const inChecked = field.type === "checkbox"
+                        ? fieldValues[field.name] === "true"
+                        : undefined;
+    const eventValue = (e: React.ChangeEvent<HTMLInputElement>) => {
+      return field.type === "checkbox" ? String(e.target.checked) : e.target.value;
+    };
+    return (
+      <input
+        id={field.name}
+        name={field.name}
+        className={inline ? "" : "full-row"}
+        onChange={e => handleValueChange(e.target.name, eventValue(e))}
+        type={inputType}
+        value={inValue}
+        checked={inChecked}
+        size={field.charSize}
+        ref={index===0 ? input1Ref : undefined}
+      />
+    );
+  }
+
+  function renderCheckbox(field: IField, index: number, inline: boolean) {
+    return (
+      // cf. https://dev.to/proticm/styling-html-checkboxes-is-super-easy-302o
+      <label className="checkbox">
+        {renderInput(field, index, inline)}
+        {field.label && <span>{field.label}</span>}
+      </label>
```

Comment:
The items inside the `<label>` might look a tad better if vertically aligned, but that's fairly trivial.

Reply:
Adjusted